### PR TITLE
fix 15438 maintain white space in display name

### PR DESCRIPTION
**Pull request by tienifr**
### Details

Maintain whitespace when showing display name

### Fixed Issues

$ https://github.com/Expensify/App/issues/15438   
PROPOSAL: https://github.com/Expensify/App/issues/15438#issuecomment-1445001066

### Tests

1. Open the app
2. Navigate to Settings > Profile
3. Click on Display Name
4. In the First Name field enter a string separated by a bunch of spaces (eg : Daraksha                  Test.)
5. Save
6. Verify the display name is correct within white spaces on setting page, profile page, LHN, report header, report item header, participant local time, IOU message, new chat page, new group page, send money page, request money page, detail page

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Open the app
2. Navigate to Settings > Profile
3. Click on Display Name
4. In the First Name field enter a string separated by a bunch of spaces (eg : Daraksha                  Test.)
5. Save
6. Verify the display name is correct within white spaces on setting page, profile page, LHN, report header, report item header, participant local time, IOU message, new chat page, new group page, send money page, request money page, detail page
- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>
1. setting page
 
Screen Shot 2023-03-04 at 00 39 42

2. profile page
 
Screen Shot 2023-03-04 at 00 40 14

3. LHN
Screen Shot 2023-03-04 at 00 40 42

4. report header
Screen Shot 2023-03-04 at 00 40 54

5. report item header
Screen Shot 2023-03-04 at 00 41 10

6. participant local time
Screen Shot 2023-03-04 at 00 41 24

7. IOU message
Screen Shot 2023-03-04 at 00 42 13

8. new chat page
Screen Shot 2023-03-04 at 00 42 36

9. new group page
Screen Shot 2023-03-04 at 00 42 58

10. send money page
Screen Shot 2023-03-04 at 00 43 30

11. request money page
Screen Shot 2023-03-04 at 00 43 51

12. detail page
Screen Shot 2023-03-04 at 00 44 06

13. Last Mesage in LHN
Screen Shot 2023-03-06 at 17 20 53

14. Subtitle in header of workspace pages

Screen Shot 2023-03-07 at 09 44 59

15. Workspace name in LHN

Screen Shot 2023-03-07 at 10 30 12

Screen Shot 2023-03-09 at 10 36 50

16. Workspace name in detail workspace chat

Screen Shot 2023-03-07 at 10 30 25

17. Workspace name in settings of workspace chat detail

Screen Shot 2023-03-07 at 10 30 43

</details>

<details>
<summary>Mobile Web - Chrome</summary>

1. setting page
image

2. profile page
image

3. LHN
image

4. report header
image

5. report item header
image
6. participant local time
image
7. IOU message
image
8. new chat page
image

9. new group page
image

10. send money page
image

11. request money page
image

12. detail page
image

13. Last Mesag
Screen Shot 2023-03-06 at 17 24 37
e in LHN

14. Subtitle in header of workspace pages

Screen Shot 2023-03-07 at 09 45 06

15. Workspace name in LHN

Screen Shot 2023-03-07 at 10 37 36

Screen Shot 2023-03-09 at 10 37 07

16. Workspace name in the subtitle of report

Screen Shot 2023-03-07 at 10 37 44

Screen Shot 2023-03-09 at 10 37 16

17. Workspace name in detail workspace chat
Screen Shot 2023-03-07 at 10 37 51

18. Workspace name in settings of workspace chat detail
Screen Shot 2023-03-07 at 10 37 58

</details>

<details>
<summary>Mobile Web - Safari</summary>

1. setting page
image

2. profile page
image

3. LHN
image

4. report header
image

5. report item header
image
6. participant local time
image
7. IOU message
image
8. new chat page
image

9. new group page
image

10. send money page
image

11. request money page
image

12. detail page
image

13. Last Mesage in LHN
Screen Shot 2023-03-06 at 17 24 37

14. Subtitle in header of workspace pages
Screen Shot 2023-03-07 at 09 45 30

14. Subtitle in header of workspace pages

Screen Shot 2023-03-07 at 09 44 59

15. Workspace name in LHN

Screen Shot 2023-03-07 at 10 37 36

Screen Shot 2023-03-09 at 10 37 07

16. Workspace name in the subtitle of report

Screen Shot 2023-03-07 at 10 37 44

Screen Shot 2023-03-09 at 10 37 16

17. Workspace name in detail workspace chat
Screen Shot 2023-03-07 at 10 37 51

18. Workspace name in settings of workspace chat detail
Screen Shot 2023-03-07 at 10 37 58

</details>

<details>
<summary>Desktop</summary>
1. setting page
 
Screen Shot 2023-03-04 at 00 39 42

2. profile page
 
Screen Shot 2023-03-04 at 00 40 14

3. LHN
Screen Shot 2023-03-04 at 00 40 42

4. report header
Screen Shot 2023-03-04 at 00 40 54

5. report item header
Screen Shot 2023-03-04 at 00 41 10

6. participant local time
Screen Shot 2023-03-04 at 00 41 24

7. IOU message
Screen Shot 2023-03-04 at 00 42 13

8. new chat page
Screen Shot 2023-03-04 at 00 42 36

9. new group page
Screen Shot 2023-03-04 at 00 42 58

10. send money page
Screen Shot 2023-03-04 at 00 43 30

11. request money page
Screen Shot 2023-03-04 at 00 43 51

12. detail page
Screen Shot 2023-03-04 at 00 44 06

13. Last Mesage in LHN
Screen Shot 2023-03-06 at 17 20 53

14. Subtitle in header of workspace pages

Screen Shot 2023-03-07 at 09 44 59

15. Workspace name in LHN

Screen Shot 2023-03-07 at 10 30 12

Screen Shot 2023-03-09 at 10 36 50

16. Workspace name in detail workspace chat

Screen Shot 2023-03-07 at 10 30 25

17. Workspace name in settings of workspace chat detail

Screen Shot 2023-03-07 at 10 30 43

</details>

<details>
<summary>iOS</summary>

1. setting page
 
Screen Shot 2023-03-04 at 01 08 50

2. profile page
 
Screen Shot 2023-03-04 at 01 09 56

3. LHN
Screen Shot 2023-03-04 at 01 10 13

4. report header
Screen Shot 2023-03-04 at 01 10 26

5. report item header
Screen Shot 2023-03-04 at 01 10 26

6. participant local time
Screen Shot 2023-03-04 at 01 10 26

7. IOU message
Screen Shot 2023-03-04 at 01 10 26

8. new chat page
Screen Shot 2023-03-04 at 01 10 52

9. new group page
Screen Shot 2023-03-04 at 01 11 07

10. send money page
Screen Shot 2023-03-04 at 01 11 27

11. request money page

Screen Shot 2023-03-04 at 01 11 49

12. detail page
Screen Shot 2023-03-04 at 01 12 01

13. Last Mesage in LHN
Screen Shot 2023-03-06 at 17 26 28

14. Subtitle in header of workspace pages

Screen Shot 2023-03-07 at 09 45 06

15. Workspace name in LHN

Screen Shot 2023-03-07 at 10 37 36

Screen Shot 2023-03-09 at 10 37 07

16. Workspace name in the subtitle of report
Screen Shot 2023-03-07 at 10 37 44

Screen Shot 2023-03-09 at 10 37 16

17. Workspace name in detail workspace chat
Screen Shot 2023-03-07 at 10 37 51

18. Workspace name in settings of workspace chat detail
Screen Shot 2023-03-07 at 10 37 58

</details>

<details>
<summary>Android</summary>

1. setting page
 
Screen Shot 2023-03-04 at 01 23 59

2. profile page
 
Screen Shot 2023-03-04 at 01 24 11

3. LHN
Screen Shot 2023-03-04 at 01 24 25

4. report header
Screen Shot 2023-03-04 at 01 24 39

5. report item header
Screen Shot 2023-03-04 at 01 24 39

6. participant local time
Screen Shot 2023-03-04 at 01 24 39

7. IOU message
Screen Shot 2023-03-04 at 01 24 39

8. new chat page
Screen Shot 2023-03-04 at 01 25 42

9. new group page
Screen Shot 2023-03-04 at 01 26 05

10. send money page
Screen Shot 2023-03-04 at 01 26 27

11. request money page
Screen Shot 2023-03-04 at 01 26 52

12. detail page
Screen Shot 2023-03-04 at 01 25 11

13. Last Mesage in LHN
Screen Shot 2023-03-06 at 17 24 37

14. Subtitle in header of workspace pages

Screen Shot 2023-03-07 at 09 45 30

15. Workspace name in LHN

Screen Shot 2023-03-07 at 10 37 36

Screen Shot 2023-03-09 at 10 37 07

16. Workspace name in the subtitle of report
Screen Shot 2023-03-07 at 10 37 44

Screen Shot 2023-03-09 at 10 37 16

17. Workspace name in detail workspace chat
Screen Shot 2023-03-07 at 10 37 51

18. Workspace name in settings of workspace chat detail
Screen Shot 2023-03-07 at 10 37 58

</details>

### CODE REVIEW

**Comment by alex-mechler** [comment]
@tienifr Can you make sure to include the fix for the LHN bug that was mentioned here as well? It was a duplicate that was not added to the original issue body on accident

**Comment by Santhosh-Sellavel** [comment]
In Workspace Name space is missing in Header
Screenshot 2023-03-07 at 1 12 23 AM

**Review by Santhosh-Sellavel — COMMENTED** [review]
Bug :https://github.com/Expensify/App/pull/15646#issuecomment-1456850942

**Comment by tienifr** [comment]
@Santhosh-Sellavel I just updated, please help to check again

Screen Shot 2023-03-07 at 09 44 59

**Comment by tienifr** [comment]
@Santhosh-Sellavel  I just updated the workspace name in all pages, please help to check again

1. Workspace Setting

Screen Shot 2023-03-07 at 09 44 59
2. LHN

Screen Shot 2023-03-07 at 10 30 12

3. Detail workspace chat

Screen Shot 2023-03-07 at 10 30 25

4. Settings in workspace chat detail

Screen Shot 2023-03-07 at 10 30 43

**Comment by alex-mechler** [comment]
Workspace names are now appearing on two lines if they are extremely long 
image

It truncates on main 
image

**Comment by tienifr** [comment]
Hi @alex-mechler Thanks for pointing this out. We have to use `white-space: pre` on TextInput with numberOfLines prop is 1. I just updated it in all places, please help to check again

Screen Shot 2023-03-08 at 15 59 50

**Comment by alex-mechler** [comment]
The workspace name is missing extra spaces in the LHN and header 
image

**Comment by Santhosh-Sellavel** [comment]
This issue really crazy one, we could easily miss a few places if we are not careful.  Will continue testing this tomorrow meanwhile please look into the above issue @tienifr, thanks!

cc: @alex-mechler

**Comment by tienifr** [comment]
Hi @alex-mechler I don't see it in the latest version of branch `fix/15438-maintain-white-space-in-display-name` Could you help to check whether you update the new code? 
cc @Santhosh-Sellavel

**Comment by tienifr** [comment]
@alex-mechler I just fixed it, please check again
In Web 
Screen Shot 2023-03-09 at 10 36 50

In Mobile 
Screen Shot 2023-03-09 at 10 37 07
Screen Shot 2023-03-09 at 10 37 16

**Comment by Santhosh-Sellavel** [comment]
Conflicts @tienifr

**Comment by Santhosh-Sellavel** [comment]
**Bug:** Missed on New room flow

Screenshot 2023-03-10 at 3 38 11 AM

File: src/pages/workspace/WorkspaceNewRoomPage.js [hunk 1]
```diff
@@ -114,7 +114,7 @@ class WorkspaceNewRoomPage extends React.Component {
         // Workspaces are policies with type === 'free'
         const workspaceOptions = _.map(
             _.filter(this.props.policies, policy => policy && policy.type === CONST.POLICY.TYPE.FREE),
-            policy => ({label: policy.name, key: policy.id, value: policy.id}),
+            policy => ({label: policy.name.replace(/ /g, '\u00a0'), key: policy.id, value: policy.id}),
```
Comment (tienifr) [inline src/pages/workspace/WorkspaceNewRoomPage.js:117]: @Santhosh-Sellavel 

> Bug: Missed on New room flow

Thanks for pointing this out. The root cause is we are using the option tag in the picker component (react-native-picker-select --> react-native-picker --> @react-native-picker/picker --> option tag) and the option tag will automatically remove white space


Screen Shot 2023-03-10 at 13 38 24

So we need to encode the white space into the &nbsp then the white space will be displayed correctly in the option tag like this

Screen Shot 2023-03-10 at 13 43 03
Screen Shot 2023-03-10 at 13 43 32

So I think this change is good. This is result 
Screen Shot 2023-03-10 at 13 45 24
Screen Shot 2023-03-10 at 13 45 32



Please tell me about your thought

Comment (tienifr) [reply src/pages/workspace/WorkspaceNewRoomPage.js:117]: Hi @Santhosh-Sellavel Could you help to take a look at my comment?

Comment (Santhosh-Sellavel) [reply src/pages/workspace/WorkspaceNewRoomPage.js:117]: This is kind of weird and hacky. Maybe should be handled separately Thoughts @alex-mechler?

Comment (alex-mechler) [reply src/pages/workspace/WorkspaceNewRoomPage.js:117]: I agree, this feels very hacky. Lets note this as a separate issue, and not handle it here.

Comment (tienifr) [reply src/pages/workspace/WorkspaceNewRoomPage.js]: @Santhosh-Sellavel I just reverted the hacky fix. Please check again and tell me there are any problem

Comment (tienifr) [reply src/pages/workspace/WorkspaceNewRoomPage.js]: Hi @Santhosh-Sellavel Could you take a look at this PR? Thanks

**Comment by Santhosh-Sellavel** [comment]
In the Archived room banner workspace name is not handled correctly
Screenshot 2023-03-17 at 12 03 19 AM

**Comment by Santhosh-Sellavel** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by tienifr** [comment]
@Santhosh-Sellavel  I just fixed that

Screen Shot 2023-03-17 at 09 13 03

**Comment by alex-mechler** [comment]
Spent some time poking around, didn't see any spots that were immediately missed. @Santhosh-Sellavel Please ping me when this is ready for my review!

**Comment by Santhosh-Sellavel** [comment]
Weird I still don't see the space
Screenshot 2023-03-20 at 12 16 41 AM

**Comment by Santhosh-Sellavel** [comment]
All Issue still not addressed on chat search results

https://user-images.githubusercontent.com/85645967/226200049-fb7f25f6-2613-4c94-bf0e-2fcd1251d98c.mov

**Comment by tienifr** [comment]
@Santhosh-Sellavel 
I just fixed the workspace name in the search result

Screen Shot 2023-03-20 at 13 14 30

We are rendering the workspace name in the Archived room banner as HTML. So I suggest adding white-space: pre to the `strong` tag. like [the lastest commit],(https://github.com/Expensify/App/pull/15646/commits/46245192880279ec22f58a2a9361b054dfe1daf2)

If we don't want to apply the white-space attribute to all strong tags, we also can create a new tag with white-space: pre and fontWeight: bold and use it in the  Archived room banner. Both solutions work well 
Screen Shot 2023-03-20 at 13 14 21

File: src/components/HTMLEngineProvider/BaseHTMLEngineProvider.js [hunk 2]
```diff
@@ -44,6 +44,10 @@ const customHTMLElementModels = {
         tagName: 'email-comment',
         mixedUAStyles: {whiteSpace: 'normal'},
     }),
+    strong: defaultHTMLElementModels.span.extend({
+        tagName: 'strong',
+        mixedUAStyles: {whiteSpace: 'pre'},
+    }),
```
Comment (Santhosh-Sellavel) [inline src/components/HTMLEngineProvider/BaseHTMLEngineProvider.js:50]: Are we good with this @alex-mechler ?

File: src/components/ArchivedReportFooter.js [hunk 3]
```diff
@@ -74,6 +74,7 @@ const ArchivedReportFooter = (props) => {
             })}
             shouldRenderHTML={archiveReason !== CONST.REPORT.ARCHIVE_REASON.DEFAULT}
             shouldShowIcon
+            textStyles={styles.pre}
```
Comment (Santhosh-Sellavel) [inline src/components/ArchivedReportFooter.js:77]: We won't need this right?

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]
Tests well, just few queries

File: src/components/ArchivedReportFooter.js (hunk 3 above)
Comment (tienifr) [reply src/components/ArchivedReportFooter.js]: That right. Thanks for pointing out that

File: src/components/HTMLEngineProvider/BaseHTMLEngineProvider.js (hunk 2 above)
Comment (alex-mechler) [reply src/components/HTMLEngineProvider/BaseHTMLEngineProvider.js:50]: I played with a variety of spacings with strong tags here and on main, and couldn't get this to break. I think this is fine. 
image
image

Comment (Santhosh-Sellavel) [reply src/components/HTMLEngineProvider/BaseHTMLEngineProvider.js:50]: Cool, thanks!

**Review by Santhosh-Sellavel — APPROVED** [review]
LGTM, Tests well!

**Comment by Santhosh-Sellavel** [comment]
Over to you @alex-mechler!

**Review by alex-mechler — APPROVED** [review]
Thanks for keeping with this!

File: src/pages/workspace/WorkspaceNewRoomPage.js (hunk 1 above)
Comment (alex-mechler) [reply src/pages/workspace/WorkspaceNewRoomPage.js]: I created a follow up issue for this. https://github.com/Expensify/App/issues/16329

**Comment by marcaaron** [comment]
This PR is causing a regression please see -> https://github.com/Expensify/App/issues/16376

**Comment by alex-mechler** [comment]
Revert PR https://github.com/Expensify/App/pull/16395

**Pull request merged**